src/core/time.ts

```diff
@@ -212,6 +212,48 @@ export class TimeValue {
             }
         } 
     }
+
```

Comment:
```suggestion
public static fromBinary(buffer: Buffer): TimeValue {
        const billion = BigInt(TimeUnit.secs);

        // To avoid overflow and floating point errors, work with BigInts.
        let bigTime = buffer.readBigUInt64LE(0);
        let bigSeconds = bigTime / billion;
        let bigNSeconds = bigTime % billion;

        return new TimeValue(Number(bigSeconds), Number(bigNSeconds));
}
```